python/pyspark/sql/connect/column.py

```diff
@@ -226,7 +225,7 @@ def substr(self, startPos: Union[int, "Column"], length: Union[int, "Column"]) -
         else:
             start_exp = startPos
 
-        return F.substr(self, start_exp, length_exp)
+        return substring(self, start_exp, length_exp)
```

Comment:
Could you fix the linter failure, @zhengruifeng ?
```
annotations failed mypy checks:
python/pyspark/sql/connect/column.py:228: error: Argument 2 to "substring" has incompatible type "Column"; expected "int"  [arg-type]
python/pyspark/sql/connect/column.py:228: error: Argument 3 to "substring" has incompatible type "Column"; expected "int"  [arg-type]
```

Reply:
sure